### ISSUE TITLE
Log parsing errors

### ISSUE THREAD

**Issue by squakez** [apache/camel-k#2299]
I guess this is something related to the latest work done on logging. When I start a new integration, I see many logs like this:
```
{"level":"info","ts":1621247425.567752,"logger":"camel-k.controller.integrationkit","msg":"Build running","request-namespace":"operator-test","request-name":"kit-c2h4bg9a6to9qcb2s580","api-version":"camel.apache.org/v1","kind":"IntegrationKit","ns":"operator-test","name":"kit-c2h4bg9a6to9qcb2s580"}
{"level":"error","ts":1621247426.6069756,"logger":"camel-k.maven","msg":"Unable to parse maven log","error":"invalid character 'I' looking for beginning of value","stacktrace":"github.com/apache/camel-k/pkg/util/maven.Run\n\tgithub.com/apache/camel-k/pkg/util/maven/maven.go:188\ngithub.com/apache/camel-k/pkg/builder.BuildQuarkusRunnerCommon\n\tgithub.com/apache/camel-k/pkg/builder/quarkus.go:186\ngithub.com/apache/camel-k/pkg/builder.buildQuarkusRunner\n\tgithub.com/apache/camel-k/pkg/builder/quarkus.go:160\ngithub.com/apache/camel-k/pkg/builder.(*builderStep).execute\n\tgithub.com/apache/camel-k/pkg/builder/steps.go:46\ngithub.com/apache/camel-k/pkg/builder.(*builderTask).Do\n\tgithub.com/apache/camel-k/pkg/builder/builder.go:125\ngithub.com/apache/camel-k/pkg/controller/build.(*scheduleRoutineAction).runBuild\n\tgithub.com/apache/camel-k/pkg/controller/build/schedule_routine.go:152"}
```
The integration ends up running correctly though.

CC @orpiske

**Comment by orpiske** [comment]
Yes it is. Where do you see that @squakez ? Is it in a test?

**Comment by squakez** [comment]
You should be able to see it by resetting the `Kit`s and building a new simple `Integration` (which will trigger a build). It happens with the snapshot operator version from `main`.

**Comment by orpiske** [comment]
> You should be able to see it by resetting the `Kit`s and building a new simple `Integration` (which will trigger a build). It happens with the snapshot operator version from `main`.

I couldn't reproduce this problem on my local environment (with k3s or kind) with a clean install. 

However, based on the messages you provided, I am suspicious of a few things.

> {"level":"info","ts":1621247425.567752,"logger":"camel-k.controller.integrationkit","msg":"Build running","request-namespace":"operator-test","request-name":"kit-c2h4bg9a6to9qcb2s580","api-version":"camel.apache.org/v1","kind":"IntegrationKit","ns":"operator-test","name":"kit-c2h4bg9a6to9qcb2s580"}

This would be caused if the operator container is outdated on the cluster. Would it be possible to rebuild and republish the operator you are using along with the `--operator-image-pull-policy Always` flag? 

```
{"level":"error","ts":1621247426.6069756,"logger":"camel-k.maven","msg":"Unable to parse maven log","error":"invalid character 'I' looking for beginning of value","stacktrace":"github.com/apache/camel-k/pkg/util/maven.Run\n\tgithub.com/apache/camel-k/pkg/util/maven/maven.go:188\ngithub.com/apache/camel-k/pkg/builder.BuildQuarkusRunnerCommon\n\tgithub.com/apache/camel-k/pkg/builder/quarkus.go:186\ngithub.com/apache/camel-k/pkg/builder.buildQuarkusRunner\n\tgithub.com/apache/camel-k/pkg/builder/quarkus.go:160\ngithub.com/apache/camel-k/pkg/builder.(*builderStep).execute\n\tgithub.com/apache/camel-k/pkg/builder/steps.go:46\ngithub.com/apache/camel-k/pkg/builder.(*builderTask).Do\n\tgithub.com/apache/camel-k/pkg/builder/builder.go:125\ngithub.com/apache/camel-k/pkg/controller/build.(*scheduleRoutineAction).runBuild\n\tgithub.com/apache/camel-k/pkg/controller/build/schedule_routine.go:152"}
```
One reason for this would be if the base image is lacking the jar files that print in JSON format. Could you, please, run the command `docker run -it apache/camel-k:1.5.0-SNAPSHOT ls /usr/share/maven/lib` and check if you have the following jars: 

* jcl-over-slf4j-1.7.29.jar
* jackson-annotations-2.9.10.jar
* jackson-core-2.9.10.jar
* jackson-databind-2.9.10.8.jar
* logback-classic-1.2.3.jar
* logback-core-1.2.3.jar
* logstash-logback-encoder-4.11.jar

Lastly, can you also check if the file `maven-slf4j-provider-*` is absent on the operator image?

**Comment by squakez** [comment]
I am running it locally (`kamel operator`) on a `minikube` against a fresh build from `main` branch. I have all the maven dependency installed locally.

**Comment by orpiske** [comment]
Thanks @squakez, I'll setup an environment with minikube and see if I can reproduce it.

**Comment by orpiske** [comment]
> I am running it locally (`kamel operator`) on a `minikube` against a fresh build from `main` branch. I have all the maven dependency installed locally.

I tried with Minikube but I couldn't reproduce the problem. 

Could you, please, run `docker run -it apache/camel-k:1.5.0-SNAPSHOT ls /usr/share/maven/lib` and let me know what you have as the output?

**Comment by squakez** [comment]
The operator is running from the local machine, not inside kubernetes. However, I've pushed the snapshot image built from `main` into the docker container and run the command to see the outcome:
```
$ docker run -it apache/camel-k:1.5.0-SNAPSHOT ls /usr/share/maven/lib
cdi-api-1.0.jar			   maven-embedder-3.6.3.jar
cdi-api.license			   maven-model-3.6.3.jar
commons-cli-1.4.jar		   maven-model-builder-3.6.3.jar
commons-cli.license		   maven-plugin-api-3.6.3.jar
commons-io-2.5.jar		   maven-repository-metadata-3.6.3.jar
commons-io.license		   maven-resolver-api-1.4.1.jar
commons-lang3-3.8.1.jar		   maven-resolver-connector-basic-1.4.1.jar
commons-lang3.license		   maven-resolver-impl-1.4.1.jar
ext				   maven-resolver-provider-3.6.3.jar
guava-25.1-android.jar		   maven-resolver-spi-1.4.1.jar
guava.license			   maven-resolver-transport-wagon-1.4.1.jar
guice-4.2.1-no_aop.jar		   maven-resolver-util-1.4.1.jar
guice.license			   maven-settings-3.6.3.jar
jackson-annotations-2.9.10.jar	   maven-settings-builder-3.6.3.jar
jackson-core-2.9.10.jar		   maven-shared-utils-3.2.1.jar
jackson-databind-2.9.10.8.jar	   org.eclipse.sisu.inject-0.3.4.jar
jansi-1.17.1.jar		   org.eclipse.sisu.inject.license
jansi.license			   org.eclipse.sisu.plexus-0.3.4.jar
jansi-native			   org.eclipse.sisu.plexus.license
javax.inject-1.jar		   plexus-cipher-1.7.jar
javax.inject.license		   plexus-cipher.license
jcl-over-slf4j-1.7.29.jar	   plexus-component-annotations-2.1.0.jar
jcl-over-slf4j.license		   plexus-component-annotations.license
jsoup-1.12.1.jar		   plexus-interpolation-1.25.jar
jsoup.license			   plexus-interpolation.license
jsr250-api-1.0.jar		   plexus-sec-dispatcher-1.4.jar
jsr250-api.license		   plexus-sec-dispatcher.license
logback-classic-1.2.3.jar	   plexus-utils-3.2.1.jar
logback-core-1.2.3.jar		   plexus-utils.license
logstash-logback-encoder-4.11.jar  slf4j-api-1.7.29.jar
maven-artifact-3.6.3.jar	   slf4j-api.license
maven-builder-support-3.6.3.jar    wagon-file-3.3.4.jar
maven-compat-3.6.3.jar		   wagon-http-3.3.4-shaded.jar
maven-core-3.6.3.jar		   wagon-provider-api-3.3.4.jar
```

**Comment by orpiske** [comment]
> The operator is running from the local machine, not inside kubernetes. However, I've pushed the snapshot image built from `main` into the docker container and run the command to see the outcome:

Thanks. So, I guess that is the problem. I believe that, in this case, you are not using the modified maven we have in the operator container.

In order for Maven to output the logs in JSON format, it has to be modified so that its original logging framework is replaced w/ one that outputs in JSON format. There are some details about the [process in the original ticket](https://github.com/apache/camel-k/issues/1783).

**Comment by squakez** [comment]
I've had a look at the original issue, but I could not find what I need to change on my side in order to fix that problem. Can you please point me at that?

I also suggest to write some documentation and when capturing the error in the logs, remanding to the documentation in order to show how to fix the problem. Otherwise, any user may be fouled to think the operator is just broken.

**Comment by astefanutti** [comment]
If I understand it correctly, this affects Camel K developers that run the operator locally, rather than users that typically run the operator using the image that contains the required Maven bits.

I'd be inclined to avoid mandating developers to patch their local Maven installation, if introducing a mechanism that falls back to unstructured Maven logging is possible, and not too convoluted.

**Comment by orpiske** [comment]
> If I understand it correctly, this affects Camel K developers that run the operator locally, rather than users that typically run the operator using the image that contains the required Maven bits.

That's correct. It only affect us (Camel K) developers running the operator outside the container. 

> I'd be inclined to avoid mandating developers to patch their local Maven installation, if introducing a mechanism that falls back to unstructured Maven logging is possible, and not too convoluted.

I'm thinking if there's a clean way for us to switch between structured and non-structured logging, to avoid patching the maven installation. There might be ... I will take a look at this later in the day.

**Comment by astefanutti** [comment]
I think it's been fixed with #2311.
